pkg/render/egressgateway/egressgateway.go

```diff
@@ -0,0 +1,438 @@
+}
+
+func (c *component) getHTTPProbe() (string, string, string) {
+	httpProbes := c.config.EgressGW.Spec.EgressGatewayFailureDetection.HTTPProbe
+	probeURLs := strings.Join(httpProbes.URLs, ",")
+	interval := fmt.Sprintf("%ds", *httpProbes.IntervalSeconds)
+	timeout := fmt.Sprintf("%ds", *httpProbes.TimeoutSeconds)
+	return probeURLs, interval, timeout
+}
+
+func (c *component) getResources() corev1.ResourceRequirements {
+	recommendedQuantity := resource.NewQuantity(1, resource.DecimalSI)
+	egw := c.config.EgressGW
+	if egw.Spec.AWS != nil && *egw.Spec.AWS.NativeIP == operatorv1.NativeIPEnabled {
+		return corev1.ResourceRequirements{
+			Limits:   corev1.ResourceList{"projectcalico.org/aws-secondary-ipv4": *recommendedQuantity},
+			Requests: corev1.ResourceList{"projectcalico.org/aws-secondary-ipv4": *recommendedQuantity},
+		}
+	}
+	return corev1.ResourceRequirements{}
+}
+
+func (c *component) getIPPools() string {
+	egw := c.config.EgressGW
+	ippools := []string{}
+	for _, ippool := range egw.Spec.IPPools {
+		if ippool.Name != "" {
+			ippools = append(ippools, ippool.Name)
+		} else if ippool.CIDR != "" {
+			ippools = append(ippools, ippool.CIDR)
+		}
+	}
+	return concatString(ippools)
+}
+
+func (c *component) getElasticIPs() string {
+	egw := c.config.EgressGW
+	if egw.Spec.AWS != nil {
+		if len(egw.Spec.AWS.ElasticIPs) > 0 {
+			return concatString(egw.Spec.AWS.ElasticIPs)
+		}
+	}
+	return ""
+}
+
+func (c *component) getHealthTimeoutDs() string {
+	egw := c.config.EgressGW
+	return fmt.Sprintf("%ds", *egw.Spec.EgressGatewayFailureDetection.HealthTimeoutDataStoreSeconds)
+}
+
+func (c *component) egwSecurityContextConstraints() *ocsv1.SecurityContextConstraints {
+	namespacedName := fmt.Sprintf("%s-%s", c.config.EgressGW.Namespace, c.config.EgressGW.Name)
+	return &ocsv1.SecurityContextConstraints{
+		TypeMeta:                 metav1.TypeMeta{Kind: "SecurityContextConstraints", APIVersion: "security.openshift.io/v1"},
+		ObjectMeta:               metav1.ObjectMeta{Name: namespacedName},
+		AllowHostDirVolumePlugin: true,
+		AllowHostIPC:             false,
+		AllowHostNetwork:         false,
+		AllowHostPID:             false,
+		AllowHostPorts:           false,
+		AllowPrivilegeEscalation: ptr.BoolToPtr(true),
+		AllowPrivilegedContainer: true,
+		FSGroup:                  ocsv1.FSGroupStrategyOptions{Type: ocsv1.FSGroupStrategyRunAsAny},
+		RunAsUser:                ocsv1.RunAsUserStrategyOptions{Type: ocsv1.RunAsUserStrategyRunAsAny},
+		ReadOnlyRootFilesystem:   false,
+		SELinuxContext:           ocsv1.SELinuxContextStrategyOptions{Type: ocsv1.SELinuxStrategyMustRunAs},
+		SupplementalGroups:       ocsv1.SupplementalGroupsStrategyOptions{Type: ocsv1.SupplementalGroupsStrategyRunAsAny},
+		Users: []string{
+			fmt.Sprintf("system:serviceaccount:%s:%s", c.config.EgressGW.Namespace, c.config.EgressGW.Name),
+		},
+		Volumes: []ocsv1.FSType{"*"},
+	}
+}
+
+func concatString(arr []string) string {
```

Comment:
I think this function is essentially reinventing `json.Marshal` for a slice, no?